### ISSUE TITLE
When clicking on a Column in the GridField, the CMS fields should be displayed in a modal popup

### ISSUE THREAD

**Issue by robbieaverill** [silverstripe/silverstripe-ckan-registry#22]
Requires #20 

When clicking on the GridField row for a column, instead of going to the GridField detail form we should open a modal popup and display the other columns there.

* Add a PHP FormFactory implementation
* Add the schema endpoint + controller for it
* Create React FormBuilder implementation
* Hook it into the GridField row click handler with entwine

**Comment by robbieaverill** [comment]
@sachajudd I notice that in https://projects.invisionapp.com/share/8GOKGM7BTR5#/screens/325469913 you have "Type" (e.g. text/number) in the GridField but not in the form. Should be a readonly attribute that is imported from a CKAN resource and is not editable by the CMS user? If so, all good - just want to check.

**Comment by sachajudd** [comment]
Yeah @robbieaverill that's been done to reduce the amount of fields in the modal as we didn't think it impacted any of the settings or provide any additional relevant information.

**Comment by brynwhyman** [comment]
The story points in this issue have been reduced from an 8 to 3, given that first implementation will be in a GridField detail form, not a modal.

A separate story will track turning this (and other detail forms) into modals at a later stage.

**Comment by robbieaverill** [comment]
@brynwhyman do you want to remove this issue from the current sprint since we won't be doing it for MVP?
